From: Danny Kukawka <danny.kukawka@bisect.de>
Date: Fri, 17 Feb 2012 05:43:26 +0000
Subject: igbvf: reset netdevice addr_assign_type if changed
Patch-mainline: v3.4-rc1
Git-commit: 067fb4c81944f8965a08cd91c0a1d66ccd09feea
References: bnc#795299 FATE#313657

Reset the state of addr_assign_type to NET_ADDR_PERM as soon as
the MAC get changed via .ndo_set_mac_address.

v2: use bitops to reset addr_assign_type

Signed-off-by: Danny Kukawka <danny.kukawka@bisect.de>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: David Chang <dchang@suse.com>
Acked-by: Benjamin Poirier <bpoirier@suse.de>
---
 drivers/net/ethernet/intel/igbvf/netdev.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/net/ethernet/intel/igbvf/netdev.c
+++ b/drivers/net/ethernet/intel/igbvf/netdev.c
@@ -1708,6 +1708,7 @@ static int igbvf_set_mac(struct net_devi
 		return -EADDRNOTAVAIL;
 
 	memcpy(netdev->dev_addr, addr->sa_data, netdev->addr_len);
+	netdev->addr_assign_type &= ~NET_ADDR_RANDOM;
 
 	return 0;
 }
